### PR TITLE
Adjust mobile zoom for levels

### PR DESCRIPTION
**Pull request by breville**
This adjusts the viewport width for regular levels from 900px to 1200px, causing mobile devices that use this viewport to "zoom out", showing more content but with smaller sizes.  While elements are smaller, our levels work on desktop much better at 1200px than 900px, and this improved usability carries over. 

We did try 1400px but it made things quite small on small devices, and didn't seem necessary for fitting all the UI elements for the scripts most often used on mobile devices.

It removes the special-casing for maze added in https://github.com/code-dot-org/code-dot-org/pull/32240.

The second [commit](https://github.com/code-dot-org/code-dot-org/pull/33983/commits/d564c59ea9cfc72fd9ec82ecbf4f0acd66004854) helped me understand what I actually did in https://github.com/code-dot-org/code-dot-org/pull/32206 for Oceans.  I thought I was setting the landscape viewport width to 600px in that change, but I was actually setting the portrait viewport width, by virtue of the scaling value generated.  That is, the document body was 600px wide in portrait, while in landscape it was significantly wider, though with a height of 600px.  

In this change we're doing something different.  We want the landscape viewport width, which is the larger of the screen's two dimensions, and the document body itself, to be 1200px, and we pick a scaling value to that effect.  When in portrait mode, we are still 1200px wide, but with the scaling value we chose, it causes the page to be significantly larger than the physical screen.

The third [commit](https://github.com/code-dot-org/code-dot-org/pull/33983/commits/89bde6e7fea501bbe3c88bcaa39f1e6d90e75ac4)  shows the header even on mobile devices.

The fourth [commit](https://github.com/code-dot-org/code-dot-org/pull/33983/commits/0187de92f7f4c29f32357c46294f1bba3412dc0f) makes a substantial improvement to the "rotate phone" overlay, by using the newer [Visual Viewport API](https://developer.mozilla.org/en-US/docs/Web/API/Visual_Viewport_API), when available, to dynamically position and resize the content of the overlay to always be in view, even as the user adjusts zoom or position on their phone.  If the API isn't available, it just uses default styling.  

It also adjusts the viewport to better suit an Android issue we saw, as described in a comment, and removes the special parameter for setting up the "rotate phone" overlay for the Oceans tutorial.

The [fifth](https://github.com/code-dot-org/code-dot-org/pull/33983/commits/00df3bdf06271131ef51218e5e1605ec66f6ce8d) commit tells iOS Safari to ignore the notch.  A downside is that the notch now covers some of our content, but an upside is that it no longer shifts some of our content off the right edge of the screen.

Here are actual iPhone 11 Pro Max screen captures, though after using iOS Safari's somewhat-hard-to-find option to make landscape view go almost full screen.  Otherwise, the user must try and scroll up, to "push" the address bar and tabs out of the way.

### before

IMG_0732

### after

IMG_0731

### Rotate overlay

On a browser that supports the Visual Viewport API, the "rotate phone" overlay consistently looks like this.  If the user pans or zooms, it will refresh itself like this 1/5 of a second later.

iphone_11_rotate

### CODE REVIEW

**Review by markabarrett — COMMENTED** [review]
If I understand the screen's right, we still don't fit all the content and everything is a little harder to touch/read?

File: apps/src/templates/RotateContainer.jsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export default class RotateContainer extends React.Component {
       // with a white background, so that if you scroll off of the inner container
       // you see white instead of the codeApp
 
-      const scale = screen.height / 900;
+      const scale = screen.height / 1400;
       width = window.screen.width / scale;
       height = window.screen.height;
```
Comment (jmkulwik) [inline apps/src/templates/RotateContainer.jsx]: Here, we make the assumption that iOS and Android measure width and height the same, but unfortunately, they don't - the width on iOS is the narrow direction and the height is the tall direction. For android, the width and height are dependent on screen orientation. i.e. on my (Android) phone, if I load in landscape mode and then turn the phone to portrait mode, I see this:


To handle this, we can use this logic:
```
      const windowHeight = Math.max(window.screen.width, window.screen.height);
      const windowWidth = Math.min(window.screen.width, window.screen.height);
      const scale = windowHeight / 1400;
      width = windowWidth / scale;
      height = windowHeight;
```

**Comment by jmkulwik** [comment]
> If I understand the screen's right, we still don't fit all the content and everything is a little harder to touch/read?

Oh hmm. I didn't notice that until you mentioned it. @breville , any idea why your viewport is shifted?

**Comment by jmkulwik** [comment]
This is a huge improvement on my phone. There is space for the angle picker on Artist now, all parts of Dance Party fit on the screen, and the buttons are now on-screen in minecraft.
Screenshot_20200401-161939_Chrome
Screenshot_20200401-160738_Chrome
Screenshot_20200401-161249_Chrome

Minecraft still felt very small (I was squinting a lot), but that shouldn't be a blocker.

**Comment by markabarrett** [comment]
Ooh la la, looks great on your device, Jessie!

**Comment by breville** [comment]
> > If I understand the screen's right, we still don't fit all the content and everything is a little harder to touch/read?
> 
> Oh hmm. I didn't notice that until you mentioned it. @breville , any idea why your viewport is shifted?

I don't yet know why this happens, but it appears to already happen prior to this change, including on our production site.  Here's dance-2019 on a real iPhone:

IMG_0636

It's difficult to scroll our entire page because we take over touch controls on so many of the surfaces, but once done, it's possible to carefully scroll the page further into view.  Here's what I was able to manage, scrolling it a bit left but leaving enough of a gap for the notch (which you don't see in the screenshot), and a bit upwards to "push" the address/tab bar out of the way, on a real iPhone:

IMG_0635

Agreed that we should work on a better responsive layout in this situation to avoid things being off screen, but I think that's orthogonal to changing the zoom level.

**Comment by breville** [comment]
I've made pretty substantial additions to this PR.  Lots of new information and screenshots in the description at the top.

**Comment by jmkulwik** [comment]
Couple additional screenshots from an Android phone (Galaxy S10) - the rotate screen image looks great. So does Maze!

Screenshot_20200403-222348_Chrome

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -1834,23 +1836,30 @@ StudioApp.prototype.fixViewportForSmallScreens_ = function(viewport, config) {
...
   }
-  var width = Math.max(minWidth, desiredWidth);
-  var scale = deviceWidth / width;
 
+  // Setting `minimum-scale=scale` means that we are unable to shrink the
+  // entire playspace area down to fit on a portrait iPhone, even though
+  // it's technically not visible because of the RotateContainer on top.
+  // But setting `maximum-scale=scale` was preventing an Android from starting
+  // in the desired zoom level in landscape, so we removed that but left
+  // `minimum-scale=scale`.
```
Comment (jmkulwik) [inline apps/src/StudioApp.js:1855]: Adding some more detail for documentation purposes: We found that `initial-scale=scale` and `minimum-scale=scale` and **not** `maximum-scale=scale` were required to make Android render correctly on initial page load while allowing users to manually zoom/resize with two-finger gestures.
Full details here: https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag

**Review by jmkulwik — APPROVED** [review]
Looks awesome. Lots of great work to make this happen :)

**Review by markabarrett — COMMENTED** [review]
Awesome work!

**Pull request merged**